lib/content_generator.rb

```diff
@@ -25,10 +25,10 @@ def self.family(location, family)
       ]
     end
 
-    def self.size(location, family, size)
-      location = LocationNameConverter.to_internal_name(location)
+    def self.size(location_id, family, size)
+      location = Location[location_id]
```

Comment:
This type will lead to N+1 issues, since you'll have a query for every size.  You would need to refactor this so you could pass in a hash of valid locations and look up the locations in the hash, so it does not cause a query per method call.  Other calls to `Location[]` and similar should be audited to make sure they do not cause N+1 query issues.

Reply:
I have solved this by passing the whole location object until here and rendering the id in `resource_creation_form`. What do you think now?

Reply:
There is something wrong with this approach and rendering that I cannot seem to solve though. The moment I pick a non default region in the UI, the private subnet list becomes disabled all together. I will try fixing it on Monday but if it persists, I might simply defer this to another PR to reduce the query count. For now, the safest option seems like passing the ID and rendering accordingly.